saleor/checkout/complete_checkout.py

```diff
@@ -607,44 +607,48 @@ def _prepare_checkout(
 ):
     """Prepare checkout object to complete the checkout process."""
     checkout = checkout_info.checkout
-    clean_checkout_shipping(checkout_info, lines, CheckoutErrorCode)
-    clean_checkout_payment(
-        manager,
-        checkout_info,
-        lines,
-        discounts,
-        CheckoutErrorCode,
-        last_payment=payment,
-    )
-    if not checkout_info.channel.is_active:
-        raise ValidationError(
-            {
-                "channel": ValidationError(
-                    "Cannot complete checkout with inactive channel.",
-                    code=CheckoutErrorCode.CHANNEL_INACTIVE.value,
-                )
-            }
+    with traced_atomic_transaction():
+        # select_for_update locks objects rows till end of transaction block
+        checkout = Checkout.objects.select_for_update().get(pk=checkout.pk)
```

Comment:
In case of checkout that is already deleted, second call will raise unhandled exception here (`.get()`)